libs-scala/crypto/src/main/scala/com/daml/crypto/MessageDigestPrototype.scala

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.crypto
+
+import com.daml.scalautil.Statement.discard
+
+import java.security.MessageDigest
+
+/*
+ * Use MessageDigest prototypes as a workaround for
+ * https://bugs.openjdk.java.net/browse/JDK-7092821, similar to Guava's
+ * workaround https://github.com/google/guava/issues/1197
+ */
+final class MessageDigestPrototype(val algorithm: String) {
+  private def createDigest: MessageDigest = MessageDigest.getInstance(algorithm)
+
+  private val prototype = createDigest
+
+  private val supportsClone: Boolean =
+    try {
+      discard(prototype.clone())
+      true
+    } catch {
+      case _: CloneNotSupportedException =>
+        false
+    }
```

Comment:
Great change @skisel-da , just wondering if we shouldn't be logging the support for clone or not. 
If somehow we end-up with an implementation that doesn't support clone, it will be more difficult to debug as there is no indication if this fix is `activated` or not.

Reply:
https://github.com/digital-asset/daml/pull/13270